### ISSUE TITLE
Download and install fail crash report

### ISSUE THREAD

**Issue by L31FY** [Tobi823/ffupdater#346]
Stacktrace:
```
java.lang.IllegalArgumentException: Failed requirement.
	at de.marmaro.krt.ffupdater.installer.ApkChecker$Companion.throwIfDownloadedFileHasDifferentSize(Unknown Source:81)
	at de.marmaro.krt.ffupdater.DownloadActivity.startDownload(SourceFile:522)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$startDownload(Unknown Source:0)
	at de.marmaro.krt.ffupdater.DownloadActivity$startDownload$1.invokeSuspend(Unknown Source:11)
	at w3.a.resumeWith(Unknown Source:8)
	at i4.l0.run(Unknown Source:102)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8741)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
	Suppressed: i4.j0: [p1{Cancelling}@f119be4, Dispatchers.Main]
Timestamp: 2023-02-17T18:05:41.860

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.5 (131) release |
| Device | SM-G998U1 (p3quew, p3q, lahaina) |
| Manufacturer | samsung (samsung) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | SWDK6720, dpi, release-keys, 1673938062000 |

**Comment by ryder203** [comment]
Stacktrace:
```
java.lang.IllegalArgumentException: Failed requirement.
	at de.marmaro.krt.ffupdater.installer.ApkChecker$Companion.throwIfApkFileIsNoValidZipFile(SourceFile:121)
	at de.marmaro.krt.ffupdater.DownloadActivity.startDownload(SourceFile:546)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$startDownload(Unknown Source:0)
	at de.marmaro.krt.ffupdater.DownloadActivity$startDownload$1.invokeSuspend(Unknown Source:11)
	at w3.a.resumeWith(Unknown Source:8)
	at i4.l0.run(Unknown Source:102)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7872)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
	Suppressed: i4.j0: [p1{Cancelling}@f6c75f6, Dispatchers.Main]
Timestamp: 2023-02-18T20:58:29.463

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.5 (131) release |
| Device | Pixel 7 (panther, panther, panther) |
| Manufacturer | google (Google) | 
| Supported ABIs | arm64-v8a |
| Android version | 13 (SDK: 33) |
| OS | abfarm-release-2004-0153, android-build, release-keys, 1673310668000 |

**Comment by ledufff** [comment]
Stacktrace:
```
java.lang.IllegalArgumentException: Failed requirement.
	at de.marmaro.krt.ffupdater.installer.ApkChecker$Companion.throwIfDownloadedFileHasDifferentSize(Unknown Source:81)
	at de.marmaro.krt.ffupdater.DownloadActivity.startDownload(SourceFile:522)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$startDownload(Unknown Source:0)
	at de.marmaro.krt.ffupdater.DownloadActivity$startDownload$1.invokeSuspend(Unknown Source:11)
	at w3.a.resumeWith(Unknown Source:8)
	at i4.l0.run(Unknown Source:102)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:211)
	at android.os.Looper.loop(Looper.java:300)
	at android.app.ActivityThread.main(ActivityThread.java:8143)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1028)
	Suppressed: i4.j0: [p1{Cancelling}@d2e1d96, Dispatchers.Main]
Timestamp: 2023-02-19T16:18:52.198

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.5 (131) release |
| Device | 21081111RG (amber_global, amber, amber) |
| Manufacturer | Xiaomi (Xiaomi) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | pangu-build-component-system-30644-kkkdm-rmqw0-rn3ml, builder, release-keys, 1672846836000 |

**Comment by ledufff** [comment]
This got fixed for me by using FFUpdater 77.7.6

**Comment by ryder203** [comment]
> This got fixed for me by using FFUpdater 77.7.6

Just updated, still crashing, even after clearing data and cache

**Comment by LeSnake04** [comment]
https://github.com/Tobi823/ffupdater/issues/346#issuecomment-1437113051

Workaround: use session installer, but you get a install dialog despite being on android 12.

**Comment by L31FY** [comment]
Stacktrace:
```
java.lang.Exception: Can't find latest update for FIREFOX_NIGHTLY.
	at de.marmaro.krt.ffupdater.storage.MetadataCache.getCachedOrFetchIfOutdated(MetadataCache.kt:33)
	at de.marmaro.krt.ffupdater.storage.MetadataCache$getCachedOrFetchIfOutdated$1.invokeSuspend(MetadataCache.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8741)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@de6e372, Dispatchers.Main]
Caused by: java.lang.Exception: Can't find latest update for FirefoxNightly.
	at de.marmaro.krt.ffupdater.app.impl.AppBase.findAppUpdateStatus(AppBase.kt:96)
	at de.marmaro.krt.ffupdater.app.impl.AppBase$findAppUpdateStatus$1.invokeSuspend(AppBase.kt:0)
	... 10 more
Caused by: java.lang.NullPointerException
	at de.marmaro.krt.ffupdater.network.mozillaci.MozillaCiJsonConsumer.updateCheck(MozillaCiJsonConsumer.kt:35)
	at de.marmaro.krt.ffupdater.network.mozillaci.MozillaCiJsonConsumer$updateCheck$1.invokeSuspend(MozillaCiJsonConsumer.kt:0)
	... 10 more
Timestamp: 2023-02-20T21:04:18.858

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.6 (132) release |
| Device | SM-G998U1 (p3quew, p3q, lahaina) |
| Manufacturer | samsung (samsung) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | SWDK6720, dpi, release-keys, 1673938062000 |         

Different failure this time

**Comment by Tobi823** [comment]
@L31FY your bug is different and should be fixed in 77.7.8

**Comment by Tobi823** [comment]
I think the other bugs are solved in 77.7.7. I suspect that FFUpdater downloaded a different files due to the old url and this file was not a valid APK file.
